src/routes/tasks.rs

```diff
@@ -59,7 +62,39 @@ pub async fn create(
         .map(|task| (Status::Created, Json(task)))
 }
 
-pub async fn show(conn: &mut Conn, task: Task) -> Result<Json<Task>, Error> {
+async fn refresh_metrics_if_needed(
+    task: Task,
+    db: Db,
+    client: Client,
+    crypter: &Crypter,
+) -> Result<Task, Error> {
+    if OffsetDateTime::now_utc() - task.updated_at <= Duration::from_secs(5 * 60) {
+        return Ok(task);
+    }
+    let aggregator = task.leader_aggregator(&db).await?;
```

Comment:
this changes the behavior from using the first party aggregator to using the leader aggregator. has that been discussed somewhere? that seems like a substantial change, especially if these metrics are ever used for billing

Reply:
If these numbers are currently only used by customers for their visibility (and not for billing purposes), is there a downside of fetching upload metrics from an externally operated leader that supports them? The first-party constraint came from our intent to use this for billing

Reply:
Agreed with @jbr. IMO for billing purposes we'd have a separate bit of business logic that determines who is the source of truth for billing data, i.e. the first party aggregator.